src/test/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheckTest.java

```diff
@@ -67,6 +67,7 @@ public void testClassWithPrivateCtorAndNestedExtendingSubclass() throws Exceptio
         final DefaultConfiguration checkConfig =
                 createModuleConfig(FinalClassCheck.class);
         final String[] expected = {
+            "10:5: " + getCheckMessage(MSG_KEY, "B"),
```

Comment:
we do not expect new violations. Can you explain why it appeared after you deleted package name from the class?

Reply:
Initially, when I replaced the package + class name with only class name. It generated a violation, I read the docs for FinalClassCheck and I thought that this violation is valid. But later I came to know that since the class has a nested class extending itself. It should not be violated.

So I have fixed it now.